### ISSUE TITLE
[Project] Self-supervised microstructure model fitting

### ISSUE THREAD

**Issue by PaddySlator** [connecthon/2022#4]
**Title**: Self-supervised microstructure model fitting
**Project Leader**: Paddy Slator, p.slator@ucl.ac.uk

**Description**: Machine learning is emerging as a powerful technique for fitting microstructure models to diffusion MRI data, but most approaches so far have utilised supervised learning. Self-supervised model fitting approaches that train directly on the data have been demonstrated for some simple diffusion MRI models, such as IVIM. In this project, we aim to apply self-supervised model fitting to a much bigger set of microstructure models, particularly those that incorporate directionality.

**Ideal participant requirements**: microstructure modelling, python programming
  
**Resources**: pytorch, [dmipy](https://github.com/AthenaEPI/dmipy), CPUs should be sufficient, initially HCP data, but would be interesting to explore connectome data

**Good first issues**: 
1. Convert some interesting microstructural models into pytorch format (i.e. differentiable form). We can use other model fitting code (e.g. [dmipy](https://github.com/AthenaEPI/dmipy)) as a useful baseline for the conversion. I already have a basic self-supervised ball-stick model fitting implementation on HCP data.
- NODDI
- standard model of white matter
- VERDICT
- anisotropic IVIM
- …?
2. Implement baseline fits using other model fitting approaches 
- dmipy
- supervised learning
3. Write “unit test” functions for testing pytorch model implementations against baseline, e.g. dmipy. We want to check that  output signals are the same for both.
4. Run tests to find the best/most robust model hyperparameters for training.
- learning rate
- dropout
- batch size
- …

**Acknowledgements**: Sean Epstein, Tobias Goodwin-Allcock, and Jason Lim (all UCL) helped develop this project idea.
